### ISSUE TITLE
Too Many Clicks For Group Settings To Save

### ISSUE THREAD

**Issue by Elshara** [UnigramDev/Unigram#2775]
**Describe the bug**
When creating a group in Unigram, by default, you cannot choose whether you want a basic or super group. It is only after you start to go and customize options, that you are forced to click save a few times for them to apply themselves.
**To Reproduce**
1. Go to 'the navigation menu'
2. Click on 'create new group'
3. Scroll down to 'add members'
4. You will notice right away, that this screen is not very accessible. You cannot select more than one member, and the same is true for group invitations.

**Expected behavior**
For you to be able to select multiple members on this screen. Or, to be able to make a group with just you as a member in it. But once you are done that, you should be taken to the group message area. Now, go to the edit button in the group. In older unigram clients, this is unlabelled for basic groups. And was renamed from Manage to edit recently. Try to do something basic in that group. Update the photo, update message permissions. Add a description, put a video avatar up. And you can't do anything. That setting does not save, like it does on mobile telegram clients. There is no ability to just save individual settings on each screen that they appear on. There are way too many screens to go through just to convert chat history to visible. It should be a regular radio button in the group general settings page. But it is not. Same goes for group permissions. Along with any other setting that does not save, until you click on the main general settings save button for the group to write the permission to telegram. Adding a photo requires 5 screens to go through just to upload it. Browse for the photo, click open. Then you have to skip that stupid photo editor, which is not accessible. Find done using object navigation in NVDA. Click on that by browsing to it with an invisible cursor using OCR. Then hit the back button, only to realize your photo didn't save. Then try to find the general settings area to click save there. And only one setting gets saved at a time. So then you have to scroll back all the way to update another setting, like enabling permissions. Only to realize, that the group no longer exists. Because it was converted silently without you knowing about it, to a super group from a basic group. So you have to close unigram, re-open it by finding it somewhere on your system. Then repeat all of what you just did if you made any changes before hand. Then you have to go through all those screens, just to tab through a bunch of useless information to update a setting. Like, the group description. It only saves, if you convert the group to a super group. But not before then. And you can't do that in unigram. The same goes for adding a video chat. All the video chat statistics are completely inaccessible in groups and channels. In the case of channels, you can't even see the list of available members if it's over 200. So the way in which groups are displayed and managed through unigram's interface is extremely flawed. It does not need to be this way. And the problem stems from a unified underlining issue. Which is, that there are too many false save buttons and too many isolated screens in unigram's interface to get anything done in a reasonable way. What I expect, is for these to completely go away. Have all the settings for the group on one single screen. Get rid of the back button. And have separate dialogs for chats, settings and so on. Rather than navigating a bunch of tabs in a single window.
**Screenshots**

**Version Info**
 - Unigram: [8.8.7433.0_x64]
 - Windows: [11 Version 22H2 build 25145.1000]

**Additional context**
Unigram needs keyboard short cuts if you are going to have to press tab 30 times to get any management issues solved in groups and channels beyond basic chat history functions. Even chat search is not accessible. This has to be improved.

**Comment by FrayxRulez** [comment]
I don't agree here.
Settings navigation in Unigram is literally the same as iOS or Android.
There isn't any "False save button". Any Save button that you click is going to send one or more requests to the Telegram servers to apply the changes to your chat.

Also, this ticket is a lot of confusion, I'm not even sure about what the problem is in the end.

**Comment by FrayxRulez** [comment]
As, again, settings navigation in Unigram is pretty much exactly the same as all other Telegram apps, I'm closing this

**Comment by Elshara** [comment]
This issue was a plethora of related issues that was made a year ago which have mostly since been fixed:
Adding members is now accessible. Back in June of 2022 it was not.
Adding groups without a member is now accessible. Back in June of 2022 it was not.
It is now easier to find the save or 'done' button when editing  a group's settings. Back in June of 2022, it was in a different location on the screen for a screen reader to scroll through to find it.
The photo editor also got some work so it is easier to find at the top area of object navigation for screen readers, so I would say this as partially fixed.
You are correct that now unigram largely operates with the same feature set as IOS or Android. However unlike Android or IOS, the following is still the case specific to Unigram.
If you upgrade a new group to a super group, this process is not a one-click action. You actually have to save a setting only accessible in a super group before the upgrade applies. Then you have to save that same setting again before it will actually save it on the server.
For example:
The only way in Unigram I can make a newly created group upgrade to a super group, is not via a checkbox. It's by changing chat history from hidden to visible. I click save in Unigram. But the change will not show up. IT just upgrades the group. I have to edit the group again and change it to visible once more before it will apply it to telegram servers.
Similarly, clicking save twice won't make this change happen on the same page. I have to navigate away from the old group's edit page, go into someone's recent chat, and then navigate back to the group for the edit or manage page to refresh the data for the new group.
This is why I say that it should be more efficient to manage groups. Non super groups cannot save descriptions, visible chat history to new members or be made public. In fact non super groups lose descriptions once they're created if one is made in a group that transitions later on to a super group from a non super group.
For example.
I make a new group and call my group Baked Goods.
I invite no members, so Baked Goods has just me inside. Before in telegram, I had to add someone. Very recently this no longer applies.
Lets say I add a photo for Baked Goods. All is well it saves fine.
Now I added a description for Baked Goods. "Serving apple pies fresh baked every day at this location."
Now I save that, it saves fine. But I can't save more than one setting at a time. If I go into a sub dialog, like changing the group from private to public, I would lose access to my description that I didn't save first if I didn't click save first. So in this way, only one setting can be saved at a time when editing settings in groups.
Sometimes you will come across a group of controls that has to be expanded like in a button, rather than a visible control on the same screen. These settings their own sub pages like privacy settings and photo. But some do not, like name and description. And yes there are multiple save buttons. So you have to know what settings will save if you navigate to a dialog if you already make changes before doing so otherwise only what changes you make in a sub dialog will save and you have to redo any changes you make to the main settings area if you navigate away from it before finding the done button in that dialog.
Overall I would call this issue mostly resolved, I have no issue with it remaining closed. But I do need to point out that it is in my mind, still a work in progress as the interface evolves which it does so frequently and seemingly in the right direction. Things that were once relevant to try to explain, are now laid out different.
Sorry for the confusion, like I say the issue is that it is not clear what gets saved and what does not on telegram servers because of too many clicks on the interface to get more than one setting dealt with at a time. Thankfully it is a lot easier to deal with now than it once was.
